core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -596,6 +623,14 @@ private long getFlagsForStateInsertion() {
       return aggregateCompletionStage.freeze();
    }
 
+   @Override
+   public void addStore(PersistenceConfiguration persistenceConfiguration) {
```

Comment:
The `disableStore` method needs to expand its locking now that a store can be added at runtime. We need to only update the interceptor chain in the writeLock to prevent concurrent updates to it.

Reply:
Actually you don't need the write lock to modify the interceptors: a) components other than PMI could be modifying it, and b) `AsyncInterceptorChainImpl.addInterceptorAfter()` does its own locking, so you don't need any locking if you anchor everything on EntryWrappingInterceptor.

Reply:
I think I will leave the lock scope as it is. This method is far from being a hot spot, will be called likely never or 1 time per cache in the cluster lifetime